pkg/arvo/tests/sys/zuse/format.hoon

```diff
@@ -0,0 +1,597 @@
+      !>  (of-wain ~['hey' '' 'there'])
+  ==
+:: join a list of lines (tapes) into a single cord.
+:: appends an extra newline - this matches unix conventions of a
+:: trailing newline. Also see #1, #2
+++  test-of-wall
+  ;:  weld
+    :: basic usage
+    %+  expect-eq
+      !>  "hey\0athere\0aworld!\0a"
+      !>  (of-wall ~["hey" "there" "world!"])
+    :: empty list
+    %+  expect-eq
+      !>  ""
+      !>  (of-wall ~)
+    :: single list
+    %+  expect-eq
+      !>  "hey\0a"
+      !>  (of-wall ~["hey"])
+    :: list with empties
+    %+  expect-eq
+      !>  "hey\0a\0athere\0a"
+      !>  (of-wall ~["hey" "" "there"])
+  ==
+:: encoding and decoding of beams <-> paths
+:: (a beam is a fully-qualified file reference. ship, desk, version,
+:: path)
+++  test-beam
+  =+  b=`beam`[[p=~zod q=%home r=[%ud p=12]] s=/hoon/zuse/sys]
+  =+  p=`path`/~zod/home/12/sys/zuse/hoon
+  ;:  weld
+    :: proper encode
+    %+  expect-eq
+      !>  p
+      !>  (en-beam b)
+    :: proper decode
+    %+  expect-eq
+      !>  (some b)
+      !>  (de-beam p)
+    :: proper round trip
+    %+  expect-eq
+      !>  (some b)
+      !>  (de-beam (en-beam b))
+    :: path too short
+    %+  expect-eq
+      !>  ~
+      !>  (de-beam /~zod/home)
+    :: invalid ship
+    %+  expect-eq
+      !>  ~
+      !>  (de-beam /'~zodisok'/home/12/sys/zuse/hoon)
+    :: invalid desk
+    %+  expect-eq
+      !>  ~
+      !>  (de-beam /~zod/12/12/sys/zuse/hoon)
+    :: invalid case
+    %+  expect-eq
+      !>  ~
+      !>  (de-beam /~zod/home/~zod/sys/zuse/hoon)
+  ==
+:: functions for creating `json` values
+++  test-enjs
+  =,  enjs
+  =+  num=`json`[%n '4200']
+  =+  str=`json`[%s 'foo']
+  =+  bool=`json`[%b &]
+  =+  array=`json`[%a ~[num str]]
+  ;:  weld
+    :: numbers
+    %+  expect-eq
+      !>  num
+      !>  (numb 4.200)
+    %+  expect-eq
+      !>  num
+      !>  (numb 0x1068)
+    %+  expect-eq
+      !>  [%n '0']
+      !>  (numb 0)
+    :: strings
+    %+  expect-eq
+      !>  str
+      !>  (tape "foo")
+    %+  expect-eq
+      :: uses of-wall, so adds the trailing newline
+      !>  [%s 'hi\0athere\0a']
+      !>  (wall ~["hi" "there"])
+    :: objects
+    %+  expect-eq
+      !>  [%o (molt ~[['foo' num]])]
+      !>  (frond 'foo' num)
+    =+  props=~[['foo' num] ['bar' bool]]
+    %+  expect-eq
+      !>  [%o (molt props)]
+      !>  (pairs props)
+    :: time - stored as integer number of milliseconds since the unix epoch
+    %+  expect-eq
+      !>   [%n '1000']
+      !>   (time ~1970.1.1..0.0.1)
+    :: ship - store ship identity as a string
+    %+  expect-eq
+      !>  [%s 'zod']
+      !>  (ship ~zod)
+  ==
+:: dejs - functions to support recursive conversion of `json` values into
+:: something else. This version crashes when used on improper input.
+:: Prefer using dejs-soft (also tested below) which returns units
+:: instead.
+++  nul  `json`~
+++  tru  `json`[%b &]
+++  num  `json`[%n ~.12]
+++  str  `json`[%s 'hey']
```

Comment:
For clarity, it might be nice to namespace your example data structures in a separate core:

```
++  ex
  |%
  ++  nul  `json`~
  ...
  --  
```

And reference them with `nul:ex` and so on.